test/functional/wallet_upgradetohd.py

```diff
@@ -110,8 +110,9 @@ def run_test(self):
 
         self.log.info("Same mnemonic, another mnemonic passphrase, no wallet passphrase, should result in a different set of keys")
         new_mnemonic_passphrase = "somewords"
-        assert node.upgradetohd(mnemonic, new_mnemonic_passphrase)
-        assert_equal(mnemonic, get_mnemonic(node))
+        assert node.upgradetohd(mnemonic[0], new_mnemonic_passphrase)
+        assert_equal(mnemonic[0], get_mnemonic(node)[0])
+        assert mnemonic[1] != get_mnemonic(node)[1]
```

Comment:
would probably make sense to make check that mnemonic passphrase not only differs from the old one but that it actually matches the new one
```suggestion
        assert_equal(mnemonic[1], new_mnemonic_passphrase)
```